src/main/java/org/openrefine/extensions/commons/utils/CommonsImportingController.java

```diff
@@ -336,14 +360,27 @@ public List<Object> getNextRowOfCells() throws IOException {
                 }
 
                 if (indexRow < files.size()) {
-                    if (mIdsColumn.contentEquals("true")) {
-                        rowsOfCells = new ArrayList<>();
-                        rowsOfCells.add(files.get(indexRow++).findValue("title").asText());
+                    rowsOfCells = new ArrayList<>();
+                    rowsOfCells.add(files.get(indexRow++).findValue("title").asText());
+
+                    if ((categoriesColumn.contentEquals("true")) && (mIdsColumn.contentEquals("true"))) {
+                        //rowsOfCells.add(getCategoriesColumn(files.get(categoriesColumnIndex++).findValue("title").asText()));
                         rowsOfCells.add("M" + files.get(mIdIndex++).findValue("pageid").asText());
 
                         return rowsOfCells;
+
+                    } else if (categoriesColumn.contentEquals("true")) {
+                        //FIXME
+                        rowsOfCells.add(getCategoriesColumn(files.get(categoriesColumnIndex++).findValue("title").asText()));
+
+                        return rowsOfCells;
+
+                    } else if (mIdsColumn.contentEquals("true")) {
+                        rowsOfCells.add("M" + files.get(mIdIndex++).findValue("pageid").asText());
```

Comment:
Intuitively you should only need one index variable (such as `indexRow`) to iterate over `files`. When adding MIDs as an additional column I would expect that the value `mIdIndex` to be the same as `indexRow` at all times, so you should be able to use only `indexRow` unless I missed something.